### ISSUE TITLE
Port vim9script's inline-function to vim8script

### ISSUE THREAD

**Issue by prabirshrestha** [vim/vim#10218]
_Instructions: Replace the template text and remove irrelevant text (including this line)_

**Is your feature request about something that is currently impossible or hard to do? Please describe the problem.**
A clear and concise description of what is hard to do. Ex. It is difficult to [...] when [...]
(If it is related to runtime files, please check their header for where to discuss enhancements.)

If using vim8script need to create a separate function and cannot use lambdas when it contains multiple statements.

```vim
function! s:timer_cb(capture...) abort
   echom a:capture
   echom "statement 1"
   echom "statement 2"
endfunction

let capture = 1
call timer_start(1000, function('s:timer_cb', [capture]))
```

**Describe the solution you'd like**
I would like to use multi statement lambda functions in vim8script.

Similar to vim9script
```vim
let capture = 1
call timer_start(1000, {x=>{
\   echom a:capture
\   echom "statement1"
\   echom "statement2"
\ })
```

or similar to lua:

```lua
vim.fn.timer_start(100, function ()
    print 'a'
    print 'b'
end)
```

vim8 could do this:

```vim
call timer_start(100, function()
    echom 'a'
    echom 'b'
endfunction)
```

*note:* I'm using vim9script style `=>` for inline-functions instead of `->` for lambdas.

**Describe alternatives you've considered**

**Additional context**

This makes writing code that captures cloures with multiple statements really hard which is extensively used in functional programming. [callbag.vim](https://github.com/prabirshrestha/callbag.vim/blob/master/autoload/callbag.vim) which is rxjs style library is very hard to write and consume.

**Comment by brammool** [comment]
[please delete the template text when writing an issue]

> If using vim8script need to create a separate function and cannot use
> lambdas when it contains multiple statements.
> 
> ```vim
> function! s:timer_cb(capture...) abort
>    echom a:capture
>    echom "statement 1"
>    echom "statement 2"
> endfunction
> 
> let capture = 1
> call timer_start(1000, function('s:timer_cb', [capture]))
> ```
> 
> **Describe the solution you'd like**
> A clear and concise description of what you want to happen.
> I would like to use multi statement lambda functions in vim8script.
> 
> Similar to vim9script
> ```vim
> let capture = 1
> call timer_start(1000, {x=>{
> \   echom a:capture
> \   echom "statement1"
> \   echom "statement2"
> \ })
> ```
> 
> or similar to lua:
> 
> ```lua
> vim.fn.timer_start(100, function ()
>     print 'a'
>     print 'b'
> end)
> ```
> 
> vim8 could do this:
> 
> ```vim
> call timer_start(100, function()
>     echom 'a'
>     echom 'b'
> endfunction)
> ```

I was going to suggest to use :vim9cmd, but while trying it out ran into
a bug.  I'll fix that first in patch 8.2.4782

With that fix you can do:

	let s:capture = 1
	vim9cmd timer_start(1000, (x) => {
	     echom s:capture
	     echom "statement1"
	     echom "statement2"
	   })

> This makes writing code that captures cloures with multiple statements
> really hard which is extensively used in functional programming.
> [callbag.vim](https://github.com/prabirshrestha/callbag.vim/blob/master/autoload/callbag.vim)
> which is rxjs style library is very hard to write and consume.

Porting the multi-line syntax back to legacy script is likely to run
into backwards compatibility problems.

You don't mention why you would not use Vim9 script for the whole thing,
that would be simpler.

-- 
Q: What is the difference between open-source and commercial software?
A: If you have a problem with commercial software you can call a phone
   number and they will tell you it might be solved in a future version.
   For open-source software there isn't a phone number to call, but you
   get the solution within a day.

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by chrisbra** [comment]
And even if this was ported, you would still have to probably add some logic to switch between the patch that implements this for legacy vim script and Vim releases before that version.
